writer.go

```diff
@@ -7,15 +7,15 @@ import (
 )
 
 func (logger *Logger) Writer() *io.PipeWriter {
-	return logger.WriterLevel(InfoLevel)
+	return logger.WriterLevel(logger.GetLevel())
 }
 
 func (logger *Logger) WriterLevel(level Level) *io.PipeWriter {
 	return NewEntry(logger).WriterLevel(level)
 }
 
 func (entry *Entry) Writer() *io.PipeWriter {
-	return entry.WriterLevel(InfoLevel)
+	return entry.WriterLevel(entry.Logger.GetLevel())
```

Comment:
Shouldn't this honour `entry.Level` if it's not `nil`?
```suggestion
	level := entry.Level
	if level == nil {
		level = entry.Logger.GetLevel()
	}
	return entry.WriterLevel(level)
```

Reply:
Not sure of this. Unless I'm missing something, I've seen in entry.go source that Level only gets populated when the log(...) method is called, which means that Level is just a copy of Logger's Level for the last time the log(...) method was called, or nil if it never happened before. One can force entry's Level to have a value, of course, but it would be lost after a call to log(...) so I don't see any benefit of setting this fields.

Regarding the Writer() code, I see that checking for the entry's value to be non-nil, otherwise going for Logger's value, makes no practical difference to the resulting value of Level. On the other hand, the addition of the "if" clause for no added value increments the cognitive complexity of the method, thus making the code unnecessarily complicated, IMHO.

Reply:
Ah, you're right.